sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -241,6 +241,7 @@ trait RuntimeReplaceable extends UnaryExpression with Unevaluable {
   override def nullable: Boolean = child.nullable
   override def foldable: Boolean = child.foldable
   override def dataType: DataType = child.dataType
+  override lazy val canonicalized: Expression = child.canonicalized
```

Comment:
I know it. However, the aliases actually are replaced by Optimizer because `RuntimeReplaceable`. So they are no harm.